src/main/scala/mesosphere/marathon/core/storage/repository/TemplateRepositoryLike.scala

```diff
@@ -0,0 +1,58 @@
+package mesosphere.marathon
+package core.storage.repository
+
+import akka.Done
+import akka.util.ByteString
+import mesosphere.marathon.Protos.ServiceDefinition
+import mesosphere.marathon.core.storage.repository.impl.TemplateRepository.VersionBucketPath
+import mesosphere.marathon.state.{AppDefinition, PathId}
+
+import scala.concurrent.Future
+
+trait TemplateRepositoryLike {
+
+  import TemplateRepositoryLike._
+
+  def init(): Future[Done]
+
+  def create(template: Spec): Future[Template]
```

Comment:
This is an odd signature, why would I pass a different type then the return type in a CRUD?